### ISSUE TITLE
num2date can no longer handle units like 'hours since 2000-01-01 0'

### ISSUE THREAD

**Issue by shoyer** [Unidata/netcdf4-python#353]
Using v1.1.5, num2date no longer works for some obscure units:
```
In [3]: netCDF4.num2date(0, 'hours since 2000-01-01 0')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-b0a23c423fd5> in <module>()
----> 1 netCDF4.num2date(0, 'hours since 2000-01-01 0')

/Users/shoyer/dev/netcdf4-python/utils.pyx in netCDF4.num2date (netCDF4.c:5158)()
    201 contains one.
    202     """
--> 203     basedate = _dateparse(units)
    204     unit = units.split()[0].lower()
    205

/Users/shoyer/dev/netcdf4-python/utils.pyx in netCDF4._dateparse (netCDF4.c:3054)()
     19         _parse_date( isostring.strip() )
     20     if year >= 100: # don't use dateutil parser for years < 100
---> 21         basedate = dparse.parse(isostring)
     22     else:
     23         basedate = datetime(year, month, day, hour, minute, second)

/Users/shoyer/miniconda/envs/xray-debug-deps/lib/python2.7/site-packages/dateutil/parser.pyc in parse(timestr, parserinfo, **kwargs)
    741         return parser(parserinfo).parse(timestr, **kwargs)
    742     else:
--> 743         return DEFAULTPARSER.parse(timestr, **kwargs)
    744
    745

/Users/shoyer/miniconda/envs/xray-debug-deps/lib/python2.7/site-packages/dateutil/parser.pyc in parse(self, timestr, default, ignoretz, tzinfos, **kwargs)
    301
    302         if res is None:
--> 303             raise ValueError("unknown string format")
    304         repl = {}
    305         for attr in ["year", "month", "day", "hour",

ValueError: unknown string format
```

On older versions of netCDF4 (1.1.4 or earlier), this worked fine:
```
In [1]: import netCDF4

In [2]: netCDF4.num2date(0, 'hours since 2000-01-01 0')
Out[2]: datetime.datetime(2000, 1, 1, 0, 0)

In [3]: netCDF4.__version__
Out[3]: '1.1.1'
```

I know this is non-standard, but netCDF files with these sorts of units are out there in the wild. For example, see this report I got with xray: https://github.com/xray/xray/issues/316

**Comment by shoyer** [comment]
I suspect this is related to https://github.com/Unidata/netcdf4-python/pull/351

**Comment by jswhit** [comment]
fix in pull request #356 (plus a regression test)

**Comment by jswhit** [comment]
pull request #356 merged, closing now...
